### DISCUSSION TITLE
Allow launching URL/App on button press

### DISCUSSION THREAD

**Discussion by dotMorten** [cgiesche/streamdeck-homeassistant#210] [Feature Requests]
I'd love to be able to launch a browser or app via a button-press, but currently you can only set it do something on Home Assistant. For instance it might show my current solar production, but when pressing it, it should take me to my solar providers dashboard.

**Comment by Hessenpower01** [comment]
No reason to do this from HA, it's available in the standard Stream Deck functionality.

**Comment by dotMorten** [comment]
Yes as a separate button. But if I have a HA button that shows for instance current solar production and I press it, I can set up what that does - I want to be able to launch a browser to show the details

**Comment by dotMorten** [comment]
Why was this moved to discussions? Are you saying a HA button can be set to also perform a browser action?

**Comment by cgiesche** [comment]
This topic (like many others) has been moved to the discussion (Feature Request) because I want to use Issues only for planned tasks or features. Otherwise, it is very difficult to keep track of which issue is a support question, which is a feature request, and which is a genuine bug.

Specifically, the topic of "Opening URL in the browser" is not planned yet. I have thought about it several times but have not come up with a really good idea on how to integrate it sensibly into the current architecture.

**Comment by stephanedupont** [comment]
It would be a very welcomed addition I think. I'm displaying the room temperature on the Stream Deck via the plugin, and it would be great if pressing on that button would open the HA url to see the detail of this entity (or any other URL we want, like an url to a page of a HA dashboard that displays temperature for all the rooms for example).

**Comment by joshmedeski** [comment]
I would also like to see this and am happy to implement it.

I'm thinking of just adding an additional input called "Open" that accepts a URL:
<img width="1232" height="831" alt="SCR-20251022-jpwe" src="https://github.com/user-attachments/assets/837f118c-246b-47ba-a48e-7c50616ac13e" />

What do you think @cgiesche?
